daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaComponent.scala

```diff
@@ -132,12 +135,35 @@ trait SchemaComponent
       }
     }
     ee
+  }
+
+  /**
+   * Elements that enclose this.
+   *
+   * If this is already an element, this still walks outward to find the
+   * next tier out.
+   */
+  final lazy val enclosingElements: Seq[ElementBase] = LV('enclosingElements) {
```

Comment:
Try moving enclosingElements, enclosingTerms, and enclosingComponents to the factory objects. Can still call them here and delegate to the factory, but there is no reason for them to reside on the instance objects.

Reply:
Changed my mind on this. Saving this cutover for next round of incremental changes to the schema compiler. This is too big a change for this change set.